src/activity_handlers.cpp

```diff
@@ -759,7 +786,22 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
         case LST_INFINITE_MAP:
             // nothing, the liquid source is infinite
             break;
-        }
+        case LST_MONSTER:
+            if( source_mon->has_effect( effect_milked ) ) {
+                current_dur = source_mon->get_effect_dur( effect_milked );
+            } if( !source_mon->has_effect( effect_milked ) ) {
+                source_mon->add_effect( effect_milked, HOURS( 6 ) );
+            } else if( HOURS( 24 ) - current_dur < HOURS( 24 ) ) {
```

Comment:
>Why does it need to be so complicated?

Because I'm awfully bad at even the simplest of algorithm.

I think I may have fixed it now.